wagtail/contrib/wagtailapi/serializers.py

```diff
@@ -2,88 +2,83 @@
-
-        # Check django fields
-        try:
-            field = obj._meta.get_field(field_name)
-
-            if field.rel and isinstance(field.rel, models.ManyToOneRel):
-                # Foreign key
-                val = field._get_val_from_obj(obj)
-
-                if val:
-                    yield field_name, OrderedDict([
-                        ('id', field._get_val_from_obj(obj)),
-                        ('meta', OrderedDict([
-                             ('type', field.rel.to._meta.app_label + '.' + field.rel.to.__name__),
-                             ('detail_url', ObjectDetailURL(field.rel.to, val)),
-                        ])),
-                    ])
-                else:
-                    yield field_name, None
-            else:
-                yield field_name, field._get_val_from_obj(obj)
-
-            continue
-        except models.fields.FieldDoesNotExist:
-            pass
-
-        # Check attributes
-        if hasattr(obj, field_name):
-            value = getattr(obj, field_name)
-            yield field_name, force_text(value, strings_only=True)
-            continue
-
-
-class WagtailSerializer(BaseSerializer):
-    def to_representation(self, instance):
-        request = self.context['request']
-        fields = self.context.get('fields', frozenset())
-        all_fields = self.context.get('all_fields', False)
-        show_details = self.context.get('show_details', False)
-        return self.serialize_object(
-            request,
-            instance,
-            fields=fields,
-            all_fields=all_fields,
-            show_details=show_details
-        )
-
-    def serialize_object_metadata(self, request, obj, show_details=False):
+from wagtail.wagtailcore import fields as wagtailcore_fields
+
+from .utils import ObjectDetailURL, URLPath, pages_for_site
+
+
+class RelatedField(relations.RelatedField):
+    def to_representation(self, value):
+        model = type(value)
+
+        return OrderedDict([
+            ('id', value.pk),
+            ('meta', OrderedDict([
+                 ('type', model._meta.app_label + '.' + model.__name__),
+                 ('detail_url', ObjectDetailURL(model, value.pk)),
+            ])),
+        ])
+
+
+class ChildRelationField(Field):
+    """
+    Child objects are part of the pages content so we nest them on the page.
+    """
+    def __init__(self, *args, **kwargs):
+        self.child_fields = kwargs.pop('child_fields')
+        super(ChildRelationField, self).__init__(*args, **kwargs)
+
+    def to_representation(self, value):
+        serializer_class = get_serializer_class(value.model, self.child_fields)
+        serializer = serializer_class()
+
+        return [
+            # We don't call to_representation here as we don't want the id or meta
+            # to be added to the child objects
+            serializer.serialize_fields(child_object)
+            for child_object in value.all()
+        ]
+
+
+class StreamField(Field):
```

Comment:
It's not obvious what these representations look like - does it make sense to docstring them or suchlike?